pinot-core/src/main/java/com/linkedin/pinot/core/common/DataFetcher.java

```diff
@@ -33,47 +35,49 @@
 public class DataFetcher {
   private static final BlockId BLOCK_ZERO = new BlockId(0);
 
-  private final Map<String, BaseOperator> _columnToDataSourceMap;
   private final Map<String, Dictionary> _columnToDictionaryMap;
   private final Map<String, BlockValSet> _columnToBlockValSetMap;
   private final Map<String, BlockValIterator> _columnToBlockValIteratorMap;
   private final Map<String, BlockMetadata> _columnToBlockMetadataMap;
 
+  // Map from MV column name to max number of entries for the column.
+  private final Map<String, Integer> _columnToMaxNumMultiValuesMap;
+
+  // Re-usable array for all dictionary ids in the block, of a single valued column
+  private final Map<String, int[]> _columnToDictIdsMap;
```

Comment:
I am not sure why you are referring _columnToDictIdsMap as a cache, it is not, it is just reusable memory. Its values are written/read within Data fetcher and within one method call.  If there's any part of code where it's values are read/written/reused across method calls, let me know and I will fix it. Please also note the following:

1. Clients should not directly need DataFetcher or DataBlockCache at all. They should work off of projection block and block val set. In fact data fetcher should be hidden within data block cache.

2. For no-dictionary support I definitely don't want DataBlockCache to have to provide any dictId's for apis that are just trying to access actual values and not dictIds.